### DISCUSSION TITLE
Preconfigurable shortcuts

### DISCUSSION THREAD

**Discussion by Spielboerg** [storybookjs/storybook#23482] [Ideas]
### Is your feature request related to a problem? Please describe.

As you can see in this older, unsolved [issue](https://github.com/storybookjs/storybook/issues/16022), the default shortcuts are not optimal, and it would be good to be able to change the default shortcuts instead of only being able to disable them completely.

### Describe the solution you'd like

Setting some or all shortcuts in the config:
`manager.js`:
```js
addons.setConfig({
  shortcuts: {
    fullScreen: ['F'],
    togglePanel: ['A'],
    panelPosition: ['D'],
    toggleNav: ['S'],
    toolbar: ['T'],
    search: ['/'],
    focusNav: ['1'],
    focusIframe: ['2'],
    focusPanel: ['3'],
    prevComponent: ['alt', 'ArrowUp'],
    nextComponent: ['alt', 'ArrowDown'],
    prevStory: ['alt', 'ArrowLeft'],
    nextStory: ['alt', 'ArrowRight'],
    shortcutsPage: ['meta', 'shift', ','],
    aboutPage: [','],
    escape: ['escape'],
    collapseAll: ['meta', 'shift', 'ArrowUp'],
    expandAll: ['meta', 'shift', 'ArrowDown'],
    'storybook/outline-outline': ['O'],
    'storybook/viewport-next': ['V'],
    'storybook/viewport-reset': ['alt', 'V'],
    'storybook/measure-addon-measure': ['M'],
  },
})
```

Perhaps on the shortcut page next to the "Restore defaults" button, which would restore the custom configuration, there could be a second "Use Storybook defaults" button. But I'm not sure if this might be too confusing for users that are not familiar with Storybook.

### Describe alternatives you've considered

Currently, we have shortcuts disabled.

### Are you able to assist to bring the feature to reality?

no

**Comment by tyleralbee** [comment]
Is there no way to customize keyboard shortcuts outside of the GUI? I'd like them to have some sort of permanence, I'm constantly having to re-bind them.
